### ISSUE TITLE
TJWK class claims to implement RFC7517 (partially) but it really doesn't implement it at all

### ISSUE THREAD

**Issue by Jens-G** [paolo-rossi/delphi-jose-jwt#54]
The unit header says:

```
/// <summary>
///   JSON Web Key (JWK) RFC implementation (partial)
/// </summary>
/// <seealso href="https://tools.ietf.org/html/rfc7517">
///   JWK RFC Document
/// </seealso>
unit JOSE.Core.JWK;

```

RFC7517 only marginally  in a side note unter [section 4.6](https://datatracker.ietf.org/doc/html/rfc7517#section-4.6) mentions PEM format at all. The whole TJWK class is basically just a simple wrapper around a TBytes which is **assumed** by other code (e.-g. function TJWS.VerifySignature: Boolean;) to be an RSA key in PEM format. 

To put it bluntly, there is nothing (not even partially) JWK specific at TJWK at all, besides the name. 

This is how an JWK would look

```
 {"kty":"RSA",
       "n":"ofgWCuLjybRlzo0tZWJjNiuSfb4p4fAkd_wWJcyQoTbji9k0l8W26mPddx
            HmfHQp-Vaw-4qPCJrcS2mJPMEzP1Pt0Bm4d4QlL-yRT-SFd2lZS-pCgNMs
            D1W_YpRPEwOWvG6b32690r2jZ47soMZo9wGzjb_7OMg0LOL-bSf63kpaSH
            SXndS5z5rexMdbBYUsLA9e-KXBdQOS-UTo7WTBEMa2R2CapHg665xsmtdV
            MTBQY4uDZlxvb3qCo5ZwKh9kG4LT6_I5IhlJH7aGhyxXFvUK-DWNmoudF8
            NAco9_h9iaGNj8q2ethFkMLs91kzk2PAcDTW9gb54h4FRWyuXpoQ",
       "e":"AQAB",
       "d":"Eq5xpGnNCivDflJsRQBXHx1hdR1k6Ulwe2JZD50LpXyWPEAeP88vLNO97I
            jlA7_GQ5sLKMgvfTeXZx9SE-7YwVol2NXOoAJe46sui395IW_GO-pWJ1O0
            BkTGoVEn2bKVRUCgu-GjBVaYLU6f3l9kJfFNS3E0QbVdxzubSu3Mkqzjkn
            439X0M_V51gfpRLI9JYanrC4D4qAdGcopV_0ZHHzQlBjudU2QvXt4ehNYT
            CBr6XCLQUShb1juUO1ZdiYoFaFQT5Tw8bGUl_x_jTj3ccPDVZFD9pIuhLh
            BOneufuBiB4cS98l2SR_RQyGWSeWjnczT0QU91p1DhOVRuOopznQ"
      }
      

```

**Comment by paolo-rossi** [comment]
Hello @Jens-G 

yes, I know! At the start I thought that supporting the generation of cryptographic keys from the base parameters would be a fairly simple task in Delphi (with the Indy SSL stuff) but then I noticed that the majority of the function needed are not imported so the task got more complicated and time consuming, so I simply moved in time the task but never got there.

Yes, I will amend the documentation and perhaps this summer I will find some time to support the JWK specification.

Ciao,
Paolo.
